src/java/org/jivesoftware/openfire/labelling/AbstractACDF.java

```diff
@@ -0,0 +1,63 @@
+package org.jivesoftware.openfire.labelling;
+
+import org.jivesoftware.openfire.XMPPServer;
+import org.jivesoftware.openfire.user.User;
+import org.jivesoftware.openfire.user.UserManager;
+import org.jivesoftware.openfire.user.UserNotFoundException;
+import org.jivesoftware.util.JiveGlobals;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xmpp.packet.JID;
+
+/**
+ * Created by dwd on 20/03/17.
+ */
+abstract public class AbstractACDF implements AccessControlDecisionFunction {
+    private static final Logger Log = LoggerFactory.getLogger(MessageInterceptor.class);
+    private String defaultUserClearance;
+    private String defaultPeerClearance;
+    private String serverClearance;
+
+    private final String PROP_PEER_PREFIX = "clearance.domain."; // + domain
+    private final String PROP_CLEARANCE_USER_DEFAULT = "clearance.user.default";
+    private final String PROP_CLEARANCE_PEER_DEFAULT = "clearance.peer.default";
+    private final String PROP_CLEARANCE_SERVER = "clearance.server";
+
+    protected AbstractACDF() {
+        this.defaultUserClearance = JiveGlobals.getProperty(PROP_CLEARANCE_USER_DEFAULT);
+        this.defaultPeerClearance = JiveGlobals.getProperty(PROP_CLEARANCE_PEER_DEFAULT);
+        this.serverClearance = JiveGlobals.getProperty(PROP_CLEARANCE_SERVER);
```

Comment:
No hard-coded defaults? Without properties, this results in null values. Do null imply some kind of default?